iotdb-core/datanode/src/test/java/org/apache/iotdb/db/storageengine/dataregion/TsFileResourceProgressIndexTest.java

```diff
@@ -218,4 +229,100 @@ public void testHybridProgressIndex() {
     Assert.assertFalse(
         hybridProgressIndex.isAfter(new RecoverProgressIndex(1, new SimpleProgressIndex(2, 21))));
   }
+
+  @Test
+  public void testProgressIndexTopologicalSort() {
+    Random random = new Random();
+    List<ProgressIndex> progressIndexList = new ArrayList<>();
+
+    int ioTProgressIndexNum = 10, peerIdRange = 3, searchIndexRange = 1000;
```

Comment:
Make all the number larger. Random number is recommanded.

Reply:
done.